feat(content): Build-A-Mine Mission String (#9637)
endless-sky/endless-sky: data/human/lusk.txt

# Copyright (c) 2023-2024 by Alex Rowan
#
# Endless Sky is free software: you can redistribute it and/or modify it under the
# terms of the GNU General Public License as published by the Free Software
# Foundation, either version 3 of the License, or (at your option) any later version.
#
# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
# PARTICULAR PURPOSE. See the GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along with
# this program. If not, see <https://www.gnu.org/licenses/>.

> mission "BAM: Transport Lusk"

Reviewer: I think Build-A-Mine is better than BAM.

> mission "BAM: Transport Lusk"

Reviewer 2: we could also call it smth like Peculiar Mining?
or Build A Mine, because I didnt see dashes being used yet in mission names?

> mission "BAM: Transport Lusk"

Reviewer: Peculiar Mining sounds great!

> mission "BAM: Transport Lusk"

Author: Yeah, I'll probably rename it to something more interesting before I submit this PR for review. At this point, it's just serving as a placeholder.

	name "Transport Lusk to <planet>"

> 	description "Transport entrepreneur Johann Lusk to a business meeting on <destination>."
> 	source
> 		attributes "deep"
> 	destination "Ada"
> #	to offer
> #		random < 10
> 	passengers 1
> 	on offer

Reviewer 3: ```suggestion
	description "Transport entrepreneur Johann Lusk to a business meeting on <destination>. He promises to pay you <credits>."
	passengers 1
	"apparent payment" 15000
	source
		attributes "deep"
	destination "Ada"
#	to offer
#		random < 10
	on offer
```
The (unpublished) order of precedence for mission syntax places `passengers` between `description` and `source`. In addition, `apparent payment` can be used to display a payment that isn't representative of the end payment.

Author: I mentioned this in another comment as well, but it turns out that `apparent payment` is overridden by the real payment in `on complete`. The code looks ahead and spills the beans about how much the player will eventually be paid. Unfortunate.

Reviewer 3: Somehow this got duplicated. Hmm.

Reviewer 3: Apparent payment should work for the description line.

Author: Turns out it did not work for the description line either. `<credits>` rendered as "<credits>" in the game, and `<payment>` rendered as "9,486 credits".

Reviewer 3: That's very weird, it contradicts what I've experienced in the past. I made a mistake putting in `<credits>` anyway.

				goto introduction
			
			label help
			`	You join the man on the floor and help him gather the messy papers. He looks up and grunts at you appreciatively as he finishes cleaning up his mess, and you both stand up.`
			label introduction
			`	"Johann Lusk," the man says, by way of greeting. He smiles at you and extends a sweaty hand, evidently expecting you to shake it.`
			choice
				`	(Shake his hand.)`
					goto shake
				`	(Wave to him instead.)`
					goto wave

			label shake
			`	You grasp the man's hand and give it a firm shake. It's clammy and a bit unpleasant, but he seems relieved that you didn't leave him hanging. You take a moment to introduce yourself as well.`
				goto pitch
			label wave
			`	There's no way you're touching that greasy hand. You give him a wave instead and introduce yourself. Looking only a little crestfallen, Johann returns your wave.`
			label pitch

> 			`	"I hope I'm not being too forward here, but you have the look of a merchant captain about you," Johann says. "I hear your type is always looking for odd jobs, and it just so happens that I have one for you." He takes a moment to slick back his hair a bit before continuing. "You see, I'm in a bit of a pickle. I'm an entrepreneur of sorts and on my way to a very important investor meeting on <destination>. My previously arranged transport has abandoned me here for some reason, and I would thus be much obliged if you were to take me there." He looks at you hopefully, then adds, "there might be a very lucrative opportunity for you if you agree to take me to where I'm going. Also, I can pay you 15,000 credits."`

Author: This doesn't work for me, because he's lying about the payment that he can actually afford. I tried using `"apparent payment" 15000` at the top of the mission, but it's overridden by the actual payment (`9486`) in the `on complete` clause.

			choice
				`"Sure, I can help you out."`
					accept
				`"Sorry, I'm not interested."`
					decline

	on visit
		dialog `You arrive on <planet>, but realize that your escort carrying Johann Lusk has not arrived yet. Better depart and wait for your escorts to enter the system.`
	on complete
		log `Transported a certain "entrepreneur" named Johann Lusk to a business meeting on Ada. He stiffed me on payment.`
		conversation
			`	As soon as you land on <planet>, Lusk is already ready to go. He thanks you profusely as he rushes out of your ship, promising to return in one hour with your payment. There's no way you can catch him without sprinting, so you instead resign yourself to watching him go and hope that he returns with your payment soon.`
			`	Three hours go by and you're just about to write off this boondoggle as a total loss. As you begin to prepare your ship to depart, you manage to spot a severely downcast Lusk reluctantly trudging toward your ship. When he finally reaches your vessel, he seems to invite himself in and walk right past you to collapse into a chair just past the airlock.`
			choice
				`	"I take it your meeting didn't go so well?"`

> 					goto rant
> 				`	"Well, 'hello' to you too."`
> 					goto rant
> 				`	"Where's my money?"`
> 					goto nopay
> 			label rant

Reviewer 3: ```suggestion
				`	"Well, 'hello' to you too."`
				`	"Where's my money?"`
					goto nopay
			label rant
```
`goto rant` isn't necessary here.

			choice
				`	"Listen, Johann. If you just pay me what you owe me, I'll be on my way."`
					goto nopay
					to display
						not "bam: demanded payment"
				`	"Uh, sure. I'll listen."`
					goto hearmore
			label nopay
			action
				set "bam: demanded payment"
			`	For the first time since you met him, Lusk looks somewhat meek. "I... don't have it. The investors at the Ada Entrepreneurial Foundation were supposed to invest in my project. I was going to use some of that money to pay you." He looks down at his feet. "Listen, if you insist on payment right now, I will give you everything that I have." He looks at you with pleading eyes.`
			choice
				`	"It's okay. We can discuss that later. What happened at your meeting?"`
					goto rant
					to display
						not "bam: ranted"
				`	"It's okay. We can discuss that later. I'll hear you out."`
					goto hearmore
					to display
						has "bam: ranted"
				`	"I'm afraid I must insist."`

> 					goto insisted
> 				`	"If you don't pay me immediately, I am calling the police."`
> 					goto insisted
> 			label insisted

Reviewer 3: ```suggestion
				`	"If you don't pay me immediately, I'm calling the police."`
```
`label insisted` doesn't appear to be necessary. It's a matter of speaking style to contract I am.

> 			action
> 				payment 9486
> 			`	The contrite look on Lusk's face immediately morphs into one of rage. "You want your money, you mewling buffoon? Fine! Here! Take it, and I hope I never see you again!" He rummages hastily through his pockets and pulls out a grimy credit chip before flinging it at your face. You react quickly enough to catch it, but by the time you look back at him, he is already rushing out the door.`
> 			`	Upon accessing the credit chip, you find that he only paid you <payment>.`
> 				decline
> 			label hearmore

Reviewer 3: ```suggestion
			action
				payment 9486
				clear "bam: ranted"
				clear "bam: demanded payment"
			`	The contrite look on Lusk's face immediately morphs into one of rage. "You want your money, you mewling buffoon? Fine! Here! Take it, and I hope I never see you again!" He rummages hastily through his pockets and pulls out a grimy credit chip before flinging it at your face. You react quickly enough to catch it, but by the time you look back at him, he is already rushing out the door.`
			`	Upon accessing the credit chip, you find that he only paid you <payment>.`
				decline
			label hearmore
			action
				clear "bam: ranted"
				clear "bam: demanded payment"
```
You'll want to clear these conditions unless they're being used in future missions.

				clear "bam: demanded payment"
			`	"Wonderful! Meet me in the spaceport in one hour. I'll tell you everything. Trust me - you're going to want to hear this." He scrambles out of his chair and rushes out of your ship, muttering something about emeralds on his way out. You take note that he still has not paid you what he promised for the trip to <planet>.`

mission "An Investment Opportunity"

> 	name "An Investment Opportunity"

Reviewer 2: The name doesnt need to b specified when its the same
